api/provider/aws/deploy_create.go

```diff
@@ -85,6 +103,16 @@ func (d *DeployProvider) renderTaskDefinition(body []byte, familyName string) (*
 		}
 	}
 
+	// If a user does not specify any requiresCompatibilities in the task definition, we default
```

Comment:
@sesh-kebab Yeah! So, when you register a task definition, AWS will run through some logic to evaluate the task definition and check its launch type compatibilities; the task definition that AWS returns after registration will contain a new `Compatibilities` field.

It seems that AWS will always check for EC2 compatibility no matter what. If a user specifies `"FARGATE"` in the task definition's `RequiresCompatibilities` field, AWS will also check for Fargate compatibility.

The idea behind this bit of logic is to keep our users informed in spite of themselves. If a user doesn't put anything in `RequiresCompatibilities`, AWS will never check for Fargate compatibility. This bit of logic just says, _"hey, the user might not know that this field exists, because they didn't supply it. let's just go ahead and gather all the compatibility information we can for the user."_ It's just a nicety.